### PR TITLE
feat: Mapbox and osm implementation

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes https://github.com/fossasia/open-event-frontend/issues/5810

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/components/public/event-map.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import classic from 'ember-classic-decorator';
+import Component from '@ember/component';
+import ENV from 'open-event-frontend/config/environment';
+
+@classic
```
Comment (iamareebjamal) [inline app/components/public/event-map.js:5]: Please always use nonclassic glimmer components

**Comment by iamareebjamal** [comment]
Please focus on OSM right now. Complete your previous PR that you closed and implement location autocomplete

**Comment by maze-runnar** [comment]
in that previous what I had implementes that was OSM or not?
https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png

**Comment by iamareebjamal** [comment]
The one you closed which was conflicting with the freelancer work

**Comment by iamareebjamal** [comment]
https://github.com/fossasia/open-event-frontend/pull/5536

**Comment by maze-runnar** [comment]
which API would be used for location autocomplete?

**Comment by maze-runnar** [comment]
@daretobedifferent18 @sachinchauhan2889 please check if changes till now are working fine, except the autocomplete part.

**Comment by maze-runnar** [comment]
@iamareebjamal how to load placesuggestion with sometext emtered in box like event filtering. it requires a page refresh at current implementation.

**Comment by iamareebjamal** [comment]
See how schedule and calendar item menu are dynamically loaded

**Comment by iamareebjamal** [comment]
There'll be a listener on input -> which will trigger an action -> action will receive current value in input field e.g "london" -> you will use loader to fetch suggestions for london using API -> after getting response, you will set it to a `tracked` variable `suggestions` -> looping over suggestions in component will display it. 

Now as the input changes, new API call will set new suggestions displaying new suggestions. Same as in any non-ember normal framework

File: app/components/explore/side-bar.js [hunk 2]
```diff
@@ -15,6 +15,8 @@ export default class SideBar extends Component {
 
   customEndDate = null;
   @tracked showFilters = false;
+
+  suggestions = null;
```
Comment (iamareebjamal) [inline app/components/explore/side-bar.js:19]: tracked and default should be empty array

**Review by sachinchauhan2889 — COMMENTED** [review]
i have tested deployment and get this:--
1) AutoComplete for location at explore page is working fine.
review 1

2) When explore page loads, OSM show too magnified image of a place(some central asia country), it seems like empty at first sight. I think it must not be this much magnified.
review 2
3) I searched for Delhi, India. OSM pick logo shows delhi, india But it was present at its previous position(i.e some central asia country).
review 3
4) Event public page is still showing google map.
review 4
5) At create event page, typing location is not giving suggestions or auto complete. it is showing this.
review 5
6) At create event page, when we choose location from map zip code is not filling and pick logo is not showing any location.
review 8
7) when we type location manually, map is showing at previous place.
review 9
8) There is no suggestions for city state or country at enter location form.
9) My previous event location was delhi, i use enter location form to change location. i choose a place from map ex. aligarh. I save event. But the location is not updated, it is still showing delhi.
review 11

**Comment by maze-runnar** [comment]
thanks @sachinchauhan2889 for detailed review. 

@iamareebjamal i have some questions.

1. >  Event public page is still showing google map.

that is part of separate PR. And that is merged. No contained in this one.

2. > I searched for Delhi, India. OSM pick logo shows delhi, india But it was present at its previous position(i.e some central asia country).

I am not so clear how to implement this. this is showing correctly on a page refresh. But how to change it dynamically? And my question is it required?

3. >  when we type location manually, map is showing at previous place.

Again I think we needed a static map. And if we needed to show this what should be the best method means when to move pointer to different place?

4. >At create event page, when we choose location from map zip code is not filling

data is not very consistent like google API so for location some data are not present.

**Comment by iamareebjamal** [comment]
Thanks @sachinchauhan2889 

@maze-runnar Let's just add autocomplete to the event wizard and merge this. Other things caan be handled in low priority later

Great work

**Comment by maze-runnar** [comment]
ok

**Comment by iamareebjamal** [comment]
Also, please remove ember-cli-gmaps in this PR as well

**Comment by maze-runnar** [comment]
ok

**Comment by iamareebjamal** [comment]
Where is mapbox used? And anyway comment out the mapbox key line, or it'll take up wrong key

**Comment by iamareebjamal** [comment]
Also remove google maps key line in .env

**Comment by iamareebjamal** [comment]
@sachinchauhan2889 Updated PR means review pending. Check automatically. I'm fixing and closing other PRs which you two are not involved in. Goal is to drop open PRs to 25 this week

**Review by sachinchauhan2889 — COMMENTED** [review]
great work @maze-runnar .
working fine. 
I see only one difference.
1) when we enter country. suppose we type Ind, country started with In suggested. And when we select India from suggestions, all suggestions disappear. But this is not happening in locations suggestions. I select from suggested locations and they donot disappears. Even i start filling any other field example ticket, Suggestions are still there below location box.

2) i donot know this is needed or not. Suggestions are not scrollable.

**Comment by mariobehling** [comment]
Due to Google Maps API error on browsers, some speakers are reporting a blank screen at different points during the sign up process. We need to get rid of Google urgently.

**Pull request merged**